sys/include/hashes/sha1.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2016 Oliver Hahm <oliver.hahm@inria.fr>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_hashes_sha1 SHA-1
+ * @ingroup     sys_hashes
+ * @brief       Implementation of the SHA-1 hashing function
+
+ * @{
+ *
+ * @file
+ * @brief       SHA-1 interface definition
+ *
+ * @author      Wei Dai and others
+ * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ */
+
+/* This code is public-domain - it is based on libcrypt
+ * placed in the public domain by Wei Dai and other contributors. */
+
+#ifndef _SHA1_H
+#define _SHA1_H
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Length of SHA-1 digests in byte
+ */
+#define SHA1_DIGEST_LENGTH  (20)
+
+/**
+ * @brief   Length of SHA-1 block in byte
+ */
+#define SHA1_BLOCK_LENGTH   (64)
+
+/**
+ * @brief SHA-1 algorithm context
+ * @internal
+ **/
+typedef struct {
+    /** internal buffer */
+    uint32_t buffer[SHA1_BLOCK_LENGTH / sizeof(uint32_t)];
+    /** buffering current state of hashing */
+    uint32_t state[SHA1_DIGEST_LENGTH / sizeof(uint32_t)];
+    /** already processed bytes */
+    uint32_t byte_count;
+    /** I have no idea what this variable is good for */
+    uint8_t buffer_offset;
```

Comment:
Internal variable used track if `buffer` is filled before proceed taking the hash this block [1].

[1] https://github.com/RIOT-OS/RIOT/pull/4701/files#diff-cf28151045a2c90554957d839d978296R105